etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2426 @@
+*
+* @name promise-function-async
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [promise-function-async]{@link https://typescript-eslint.io/rules/promise-function-async}
+*
+* @example
+* // Okay...
+* function fetchWithoutAsync() {
+*   return fetch( 'url' ).then( response => response.json() );
+* }
+*/
+rules[ '@typescript-eslint/promise-function-async' ] = 'off';
+
+/**
+* Enforces the consistent use of single quotes for string literals and allows avoiding escape for quotes.
+*
+* @name quotes
+* @memberof rules
+* @type {Array}
+* @default [ 'error', 'single', { 'avoidEscape': true } ]
+* @see [quotes]{@link https://typescript-eslint.io/rules/quotes}
+*
+* @example
+* // Bad...
+* const bad = "bad";
+*
+* // Good...
+* const good = 'good';
+* const goodEscape = "good's";
+*/
+rules[ '@typescript-eslint/quotes' ] = [
+	'error',
+	'single',
+	{
+		'avoidEscape': true
+	}
+];
+
+/**
+* Requires that `async` functions have an `await` expression.
+*
+* @name require-await
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [require-await]{@link https://typescript-eslint.io/rules/require-await}
+*
+* @example
+* // Bad...
+* async function bad() {
+*   return 42;
+* }
+*
+* // Good...
+* async function good() {
+*   return await someAsyncFunction();
+* }
+*/
+rules[ '@typescript-eslint/require-await' ] = 'error';
+
+/**
+* When adding two variables, operands must both be of type number or of type string.
+*
+* @name restrict-plus-operands
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [restrict-plus-operands]{@link https://typescript-eslint.io/rules/restrict-plus-operands}
+*
+* @example
+* // Bad...
+* const bad = 'The total is: ' + 42;
+*
+* // Good...
+* const good = 'The total is: ' + String(42);
+*/
+rules[ '@typescript-eslint/restrict-plus-operands' ] = 'error';
+
+/**
+* Enforces the use of semicolons after statements.
+*
+* @name semi
+* @memberof rules
+* @type {Array}
+* @default [ 'error', 'always' ]
+* @see [semi]{@link https://typescript-eslint.io/rules/semi}
+*
+* @example
+* // Bad...
+* const bad = 'no semi'
+*
+* // Good...
+* const good = 'semi';
+*/
+rules[ '@typescript-eslint/semi' ] = [ 'error', 'always' ];
+
+/**
+* Disables the rule for strict boolean expressions.
+*
+* @name strict-boolean-expressions
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [strict-boolean-expressions]{@link https://typescript-eslint.io/rules/strict-boolean-expressions}
+*
+* @example
+* // Okay...
+* if ( someValue ) {}
+*/
+rules[ '@typescript-eslint/strict-boolean-expressions' ] = 'off';
+
+/**
+* Enforces triple slash references (`/// <reference path="" />`) usage.
+*
+* @name triple-slash-reference
+* @memberof rules
+* @type {Array}
+* @default [ 'error', { 'path': 'always', 'types': 'prefer-import', 'lib': 'always' } ]
+* @see [triple-slash-reference]{@link https://typescript-eslint.io/rules/triple-slash-reference}
+*
+* @example
+* // Bad...
+* /// <reference types="something" />
```

Comment:
@Planeshifter I believe we regularly violate this rule. Namely,

```javascript
/// <reference types="@stdlib/types"/>
```

Reply:
Updated the example.